core_v2/sources/v2_price_model.move

```diff
@@ -0,0 +1,121 @@
+module aptos_names_v2::v2_price_model {
+    use aptos_names_v2::v2_config;
+    use aptos_std::math64;
+    use std::error;
+
+
+    /// The domain length is too short- currently the minimum is 2 characters
+    const EDOMAIN_TOO_SHORT: u64 = 1;
+    const SECONDS_PER_YEAR: u64 = 60 * 60 * 24 * 365;
+
+    /// The longer the name is registered for, the more expensive it is per year.
+    /// The curve is exponential, with every year costing more than the previous
+    fun scale_price_for_years(price: u64, years: u8): u64 {
+        // TODO: THIS WHOLE FUNCTION IS A PLACEHOLDER
+        let final_price = 0;
+        let multiplier = 100;
+        let years = (years as u64);
+        let i = 1;
+        while (i <= years) {
+            final_price = final_price + (price * multiplier) / 100;
+            multiplier = multiplier + 5 + i * 5;
+            i = i + 1;
+        };
+        final_price
+    }
+
+    /// There is a fixed cost per each tier of domain names, from 2 to >=6, and it also scales exponentially with number of years to register
+    public fun price_for_domain(domain_length: u64, registration_secs: u64): u64 {
+        assert!(domain_length >= 2, error::out_of_range(EDOMAIN_TOO_SHORT));
+        let length_to_charge_for = math64::min(domain_length, 6);
+        let registration_years = (registration_secs / SECONDS_PER_YEAR as u8);
+        scale_price_for_years(v2_config::domain_price_for_length(length_to_charge_for), registration_years)
+    }
+
+    /// Subdomains have a fixed unit cost
+    public fun price_for_subdomain(_registration_duration_secs: u64): u64 {
+        v2_config::subdomain_price()
+    }
+
```

Comment:
I think we should just eliminate this as well since subdomains will be free from here on out...